transport/transporttest/context.go

```diff
@@ -103,7 +104,7 @@ func (c *ContextMatcher) Matches(got interface{}) bool {
 
 	if c.baggage != nil {
 		headers := baggage.FromContext(ctx)
-		if !reflect.DeepEqual(c.baggage, headers) {
+		if !reflect.DeepEqual(*c.baggage, headers) {
```

Comment:
can baggage ever be `nil`? If so, do:
```go
if !reflect.DeepEqual(c.baggage, &headers) {
```

if it should never be `nil`, this is fine, as it'll cause a panic and we'll notice very quickly